util/tippingContractUtil.js

```diff
@@ -94,27 +96,30 @@ const aggregateState = (returnState) => {
         break;
       case 'DirectAeTip':
         data = tipData[0];
+        data.type = tipType;
         data.receiver = tipData[1];
         data.amount = tipData[2];
         break;
       case 'DirectTokenTip':
         data = tipData[0];
+        data.type = tipType;
         data.receiver = tipData[1];
         data.token = tipData[2].token;
         data.token_amount = tipData[2].amount;
         data.amount = 0;
         break;
       case 'PostWithoutTip':
         data = tipData[0];
+        data.type = tipType;
         data.media = tipData[1];
         data.amount = 0;
         break;
       default:
         data = tipTypeData; // Fallback for old contract state format
+        data.type = 'AeTip';
         break;
     }
 
-    data.type = tipType ? tipType : 'AeTip';
```

Comment:
whats wrong with this?

Reply:
it didn't work for "v1" contracts, then tipType was "amount" so not being replaced by that check